app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -842,7 +842,14 @@ private void moveOverlayUpOrDownToMatchAppraisalBox() {
             return;
         }
 
-        moveOverlay(appraisalBox.getVisibility() == View.VISIBLE);
+        if (inputBox.getVisibility() == View.VISIBLE) {
+            boolean appraisalExpanded = appraisalBox.getVisibility() == View.VISIBLE;
+            moveOverlay(appraisalExpanded);
+        } else { //always move overlay down if not on input screen. Appraisal box cant be open if user is not in
+            // input screen
+            moveOverlay(false);
+        }
+
```

Comment:
Nope, because `&&` is short-circuiting — the second would be tested only if the first is true. It's just more concise.